### DISCUSSION TITLE
Pika Reader compatibility with Zebra FX7400

### DISCUSSION THREAD

**Discussion by grizzlypineracing** [PikaTimer/PikaReader#3] [General]
I posted this on Facebook and was asked to share it here. I would like to find out more information about using a Raspberry Pi (or any micro computer) as a data server (?) for a timing system using older FX7400 readers. I know that Pika Reader is being developed to link Pika Timer to Impinj style readers with the hope that it will also be able to communicate with zebra/motorola branded readers in the near future. I am just looking for advice, ideas or suggestions.

**Comment by segfaultcoredump** [comment]
The FX series is on the roadmap and something I hope to add. 

The "best" way to add support is to use the official Zebra FX series java SDK available at [https://www.zebra.com/us/en/support-downloads/software/rfid-software/host-rfid-sdk-for-c-and-java-on-linux.html?downloadId=5ed10e02-d81b-4831-a0f0-f72dbdd47ee1](https://www.zebra.com/us/en/support-downloads/software/rfid-software/host-rfid-sdk-for-c-and-java-on-linux.html?downloadId=5ed10e02-d81b-4831-a0f0-f72dbdd47ee1). The problem is that this SDK uses java JNI for, er, reasons. So it will only work with a handful of linux distros and only on x64 hardware, not ARM based systems like Raspberry PI.  

The backup plan is to poke around and figure out what the corresponding LLRP commands and use those via the old org.llrp libraries that are native java. 

I don't have an FX series reader (yet), so I figured I'd focus on the rest of the app since I was able to get the Impinj series working via their official SDK. Once that was up and running my goal was to procure an FX series reader and get that working. 

Alternatively, if somebody wanted to look at the outline of the Impinj code at https://github.com/PikaTimer/PikaReader/tree/master/src/main/java/com/pikatimer/pikareader/readers/impinj and work out the equivalent for the Zebra, they are more than welcome to do so and then contribute it to the project.
